cluster/engine.go

```diff
@@ -52,10 +47,21 @@ func (d *delayer) Wait() <-chan time.Time {
 	d.l.Lock()
 	defer d.l.Unlock()
 
-	waitPeriod := int64(d.rangeMin) + d.r.Int63n(int64(d.rangeMax)-int64(d.rangeMin))
+	waitPeriod := int64(d.rangeMin)
+	if delta := int64(d.rangeMax) - int64(d.rangeMin); delta > 0 {
+		// Int63n panics if the parameter is 0
+		waitPeriod += d.r.Int63n(delta)
+	}
 	return time.After(time.Duration(waitPeriod))
 }
 
+// EngineOpts represents the options for an engine
+type EngineOpts struct {
```

Comment:
If we're going to add this type, do we want to add the other arguments to `NewEngine` to this type? This is very specific to timing rather than the engine.